pkg/controllers/leaderworkerset_controller.go

```diff
@@ -262,6 +340,9 @@ func (r *LeaderWorkerSetReconciler) rollingUpdateParameters(ctx context.Context,
 	// Case 3:
 	// In normal cases, return the values directly.
 	if rollingUpdateCompleted {
+		if err := r.deleteMultipleHeadlessSevices(ctx, lws); err != nil {
```

Comment:
I don't think we will need this if we set the ownerReference on the headless services correctly

Reply:
Done